sdk/core/System.ClientModel/src/ModelReaderWriter/SequenceBufferReader.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.IO;
+using System.Threading.Tasks;
+using System.Threading;
+using System.Buffers;
+using System.Diagnostics;
+
+namespace System.ClientModel.Internal;
+
+internal class SequenceBufferReader : IDisposable
+{
+    private readonly SequenceBuffer[] _buffers;
+    private readonly int _count;
+    private volatile int _isDisposed;
+    private volatile int _copyCount;
+
+    public SequenceBufferReader(SequenceBuffer[] buffers, int count)
+    {
+        _buffers = buffers;
+        _count = count;
+    }
+
+    public bool TryComputeLength(out long length)
+    {
+        if (_isDisposed == 1)
+        {
+            throw new ObjectDisposedException(nameof(SequenceBufferReader));
```

Comment:
Should this return false instead?

Reply:
I believe the convention is to throw object disposed since this is not an "expected" path for this method.  Returning false is for "expected" paths?

Let me know if my understanding here isn't correct.

Reply:
The other implied question here is - should this method be called Try* and return bool if it only ever returns `true` or throws?

Reply:
thats a good question, I think this is an artifact of an old implementation where this original was inheriting from RequestContent which defines it as Try.

Since this is decoupled at this point we can probably change it to just `Length` since we for sure can calculate this every time